### PR TITLE
PHP 7 Compatibility and installer fix

### PR DESCRIPTION
**Pull request by xaoseric**
This adds php 7 compatability and fixes the installer

### CODE REVIEW

**Comment by xaoseric** [comment]
Added fix for invoice management page as well

File: install/index.php [hunk 1]
```diff
@@ -18,8 +18,8 @@
  */
 
 // The new version of THT we're installing
-define("NVER", "1.3");
-define("NVERCODE", 1010300);
+define("NVER", "1.4.0");
+define("NVERCODE", 1010400);
```
Comment (kmark) [inline install/index.php]: v1.3 was never released. Any particular reason for jumping to v1.4?

File: includes/compiler.php [hunk 2]
```diff
@@ -219,15 +219,15 @@ function checkForDependencies() {
 	// Here, we're going to see if we have the functions that we need. :D
 	$needed = array();
 	// First things first:
-	if(version_compare(PHP_VERSION, '5.3.0', '<')) {
+	if(version_compare(PHP_VERSION, '7.0.0', '<')) {
```
Comment (kmark) [inline includes/compiler.php]: This makes >= PHP 7.0 a requirement. Considering what I assume to be the very limited and somewhat legacy user base (for good reason) of this version of THT this change would hurt adoption even further. A requirement of PHP 7 (or 7.1, really) for THT2 makes sense but here I'm not so sure. 5.6 would probably be the best tradeoff.

File: includes/compiler.php [hunk 3]
```diff
@@ -219,15 +219,15 @@ function checkForDependencies() {
...
 		die("PHP Version 5.3 or greater is required! You're currently running PHP " . PHP_VERSION);
```
Comment (kmark) [inline includes/compiler.php]: This would also need to be updated to reflect the new minimum PHP version.

File: includes/compiler.php (hunk 2 above)
Comment (xaoseric) [inline includes/compiler.php]: Yeah, 5.6 would be a good trade off. I'll make it compatible with 5.6/7.0.

File: install/index.php (hunk 1 above)
Comment (xaoseric) [inline install/index.php]: The rss classes had to be removed, and a lot of the code had to be updated to be compatible with mysqli, I can change it back to 1.3.0 if you want.

File: includes/compiler.php (hunk 3 above)
Comment (xaoseric) [inline includes/compiler.php]: Will update it to reflect the new minimum PHP version.

**Comment by xaoseric** [comment]
The aim of ’init.php’ is to initialize THT, I can put the code at the top of ’compiled.php’ though.

File: install/index.php (hunk 1 above)
Comment (kmark) [inline install/index.php]: Nope, thanks for the explanation. Moving to 1.4 can make sense given I've already established (years ago) what 1.3 was supposed to be. We'll just skip that release.

**Comment by kmark** [comment]
Putting the init code into the compiler is probably best unless there's a specific reason not to. The compiler (admittedly not necessarily the best name for it) is essentially doing the initialization work as it is.

**Comment by xaoseric** [comment]
yeah, I'll look at changing some things around with the compiler and installer to make a lock file if THT is installed

**Comment by kmark** [comment]
Sounds good. Once you've updated this PR I'll review again and merge or comment.

**Comment by xaoseric** [comment]
The requested changes have been made

**Pull request merged**